src/AstTypes.h

```diff
@@ -29,6 +29,6 @@ typedef int64_t AstDomain;
 
 /** Lower and upper boundaries for the AST domain. The range must be able to be stored in a RamDomain. By
  *  default, RamDomain is uint32_t and the allowed AstDomain range is that of a int32_t. **/
-#define MIN_AST_DOMAIN (std::numeric_limits<int32_t>::min())
-#define MAX_AST_DOMAIN (std::numeric_limits<int32_t>::max())
+#define MIN_AST_DOMAIN (std::numeric_limits<AstDomain>::min())
```

Comment:
With 32 bit domains we have the luxury of reading in 64 bit numbers and checking if they are in the 32 bit range, which helps with nicer error messages. This doesn't work if the min/max values actually are the min/max value for the type.

The quick fix is to change this file so AstDomain is always int64_t (the maximum size for a while yet) and make the size used in min/max configurable. Error messages if you try to read an integer > 64bit may be unhelpful, though.

Reply:
I just saw this in the test results -- pushed a fix. Sorry about that; I missed running tests with the default 32-bit configuration. `make check` now passes for semantic tests, including the `number_constants` prior failure on Travis.

Reply:
In fact, it's the min/max of a RamDomain so this file could include RamTypes.h and get MIN/MAX_AST_DOMAIN from the size of a RamDomain.

Reply:
Yup, that's more canonical than my definition, I agree. Changed as suggested. Thanks!